lib/teiserver/player/tachyon_handler.ex

```diff
@@ -301,6 +321,172 @@ defmodule Teiserver.Player.TachyonHandler do
+      end)
+
+    incoming =
+      Account.list_incoming_friend_requests(state.user.id)
+      |> Enum.map(fn req ->
+        %{from: to_string(req.from_user_id), sentAt: NaiveDateTime.diff(req.inserted_at, epoch)}
+      end)
+
+    outgoing =
+      Account.list_outgoing_friend_requests(state.user.id)
+      |> Enum.map(fn req ->
+        %{to: to_string(req.to_user_id), sentAt: NaiveDateTime.diff(req.inserted_at, epoch)}
+      end)
+
+    resp = %{
+      friends: friends,
+      incomingPendingRequests: incoming,
+      outgoingPendingRequests: outgoing
+    }
+
+    {:response, cmd_id, resp, state}
+  end
+
+  def handle_command("friend/sendRequest" = cmd_id, "request", message_id, msg, state) do
+    with {:ok, target} <- get_user(msg["data"]["to"]),
+         {:ok, %Account.FriendRequest{}} <-
+           Account.create_friend_request(state.user.id, target.id) do
+      Player.Session.friend_request_received(target.id, state.user.id)
+      {:response, cmd_id, nil, state}
+    else
+      {:error, :invalid_user} ->
+        resp =
+          Schema.error_response(cmd_id, message_id, :invalid_user)
+          |> Jason.encode!()
+
+        {:reply, :ok, {:text, resp}, state}
+
+      # this is a bit scuffed and could be refactored so that `create_friend_request`
+      # returns an atom instead of a raw string
+      {:error, err} when is_binary(err) ->
+        resp =
+          Schema.error_response(cmd_id, message_id, :invalid_user, err)
+          |> Jason.encode!()
+
+        {:reply, :ok, {:text, resp}, state}
+
+      err ->
+        Logger.error("cannot create friend request #{inspect(err)}")
+
+        resp =
+          Schema.error_response(cmd_id, message_id, :internal_error)
+          |> Jason.encode!()
+
+        {:reply, :ok, {:text, resp}, state}
+    end
+  end
+
+  def handle_command("friend/acceptRequest" = cmd_id, "request", message_id, msg, state) do
+    with {:ok, originator_id} <- parse_user_id(msg["data"]["from"]),
+         :ok <- Account.accept_friend_request(originator_id, state.user.id) do
+      Player.Session.friend_request_accepted(originator_id, state.user.id)
+      {:response, cmd_id, nil, state}
+    else
+      {:error, :invalid_id} ->
+        resp =
+          Schema.error_response(cmd_id, message_id, :invalid_user)
+          |> Jason.encode!()
+
+        {:reply, :ok, {:text, resp}, state}
+
+      {:error, "no request"} ->
+        resp =
+          Schema.error_response(cmd_id, message_id, :no_pending_request)
+          |> Jason.encode!()
+
+        {:reply, :ok, {:text, resp}, state}
+    end
+  end
+
+  def handle_command("friend/rejectRequest" = cmd_id, "request", message_id, msg, state) do
+    with {:ok, originator_id} <- parse_user_id(msg["data"]["from"]),
+         %Account.FriendRequest{} = req <-
+           Account.get_friend_request(originator_id, state.user.id),
+         {:ok, _changeset} <- Account.delete_friend_request(req) do
```

Comment:
Why not use `Account.decline_friend_request`?

Reply:
didn't see it, amended the commit to use that function.